pkg/statedb/types.go

```diff
@@ -108,6 +108,20 @@ type Table[Obj ObjectConstraints[Obj]] interface {
 	// Writer when given a write transaction returns a table writer
 	// that can be used to modify the table.
 	Writer(tx WriteTransaction) TableReaderWriter[Obj]
+
+	// Initialized returns a channel that is closed when the table
+	// has been fully initialized. This method can only be used
+	// from start onwards.
+	Initialized() <-chan struct{}
+
+	// RegisterInitializer registers an initializer. Returns
```

Comment:
good to clarify what an initializer is here. I guess it's something which runs before the table is considered ready for use?

Reply:
It depends on use-case. Some readers might not care with the table being fully synchronized with an external data source and some might.